libs/sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -1767,7 +1767,17 @@ public Node visitDefinedDataType(SqlBaseParser.DefinedDataTypeContext context) {
         return new ColumnType(type);
     }
 
-    private String getObjectType(Token type) {
+    @Override
+    public Node visitParametrizedDataType(SqlBaseParser.ParametrizedDataTypeContext context) {
+        var name = ((ColumnType<?>) visit(context.dataType())).name();
```

Comment:
I think we can get rid of ColumnType and its derived classes and just for a string that represents the data type, then   in the analyzer we can use the type parameter signature parsing logic to get a CrateDB internal data type. I'll check that and follow up on it if it is possible after adding the `varchar(length)` support.